web/src/components/modals/standardize-metadata.tsx

```diff
@@ -0,0 +1,346 @@
+      const [key, ...values] = row;
+      obj[key] = values;
+      return obj;
+    }, {});
+
+  const [selectedOption, setSelectedOption] = useState<AvailableSchemas>('');
+  const [selectedValues, setSelectedValues] = useState<SelectedValues>({});
+  const [whereDuplicates, setWhereDuplicates] = useState<number[] | null>(null);
+
+  const {
+    isFetching,
+    isError,
+    error,
+    data,
+    refetch: standardize,
+  } = useStandardize(namespace, project, tag, selectedOption?.value);
+
+  const standardizedData = data?.results;
+
+  const formatToPercentage = (value: number): string => {
+    return `${(value * 100).toFixed(2)}%`;
+  };
+
+  const handleRadioChange = (key, value) => {
+    setSelectedValues((prev) => {
+      const newValues = {
+        ...prev,
+        [key]: value === null ? key : value,
+      };
+
+      setWhereDuplicates(checkForDuplicates(newValues));
+
+      return newValues;
+    });
+  };
+
+  const checkForDuplicates = (values) => {
+    const valueArray = Object.values(values);
+    const duplicates = {};
+    const result = [];
+
+    for (let i = 0; i < valueArray.length; i++) {
+      const value = valueArray[i];
+      if (value in duplicates) {
+        // If we haven't added this value's index yet, add it
+        if (duplicates[value].length === 1) {
+          result.push(duplicates[value][0]);
+        }
+        result.push(i);
+      } else {
+        duplicates[value] = [i];
+      }
+    }
+
+    return result.length > 0 ? result : null;
+  };
+
+  const getDefaultSelections = (standardizedData) => {
+    const defaultSelections = {};
+    Object.keys(standardizedData).forEach((key) => {
+      defaultSelections[key] = key;
+    });
+    return defaultSelections;
+  };
+
+  const updateTabDataRaw = (tabDataRaw: TabData, selectedValues: SelectedValues): TabData => {
+    if (tabDataRaw.length === 0) return tabDataRaw;
+
+    // Create a new array to avoid mutating the original
+    const updatedTabDataRaw: TabData = [tabDataRaw[0].slice(), ...tabDataRaw.slice(1)];
+
+    // Update only the column names (first row) based on selectedValues
+    Object.entries(selectedValues).forEach(([key, value]) => {
+      const columnIndex = updatedTabDataRaw[0].indexOf(key);
+      if (columnIndex !== -1 && key !== value) {
+        updatedTabDataRaw[0][columnIndex] = value;
+      }
+    });
+
+    return updatedTabDataRaw;
+  };
+
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    // Refetch data with new selectedOption
+    await standardize();
+  };
+
+  const prepareHandsontableData = useCallback((key: string) => {
+    const selectedValue = selectedValues[key] || '';
+    const topValues = tabData[key]?.slice(0, 6).map((item) => [item]) || [];
+    const emptyRows = Array(Math.max(0, 6 - topValues.length)).fill(['']);
+
+    return [[selectedValue], ...topValues, ...emptyRows];
+  }, [selectedValues, tabData]);
+
+  useEffect(() => {
```

Comment:
I'm trying to think of ways to get rid of this `useEffect`... I might need to pull the code and look at it in detail